tck/src/main/java/org/eclipse/microprofile/config/tck/ClassConverterTest.java

```diff
@@ -0,0 +1,110 @@
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.eclipse.microprofile.config.tck.base.AbstractTest.addFile;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * Test Class type converter handling
+ *
+ * @author John D. Ament
+ */
+public class ClassConverterTest extends Arquillian {
+
+    @Deployment
+    public static WebArchive deploy() {
+        String archiveName = ClassConverterTest.class.getSimpleName();
+        JavaArchive testJar = ShrinkWrap
+                .create(JavaArchive.class, archiveName+".jar")
+                .addClass(ClassConverterBean.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .as(JavaArchive.class);
+
+            addFile(testJar, "META-INF/microprofile-config.properties");
+
+            WebArchive war = ShrinkWrap
+                .create(WebArchive.class, archiveName+".war")
+                .addAsLibrary(testJar);
+            return war;
+    }
+
+    @Inject
+    private Config config;
+
+    @Inject
+    private ClassConverterBean classConverterBean;
+
+    @Test
+    public void testClassConverterWithLookup() {
+        Class<?> testClass = config.getValue("tck.config.test.javaconfig.converter.class", Class.class);
+        assertEquals(testClass, ClassConverterTest.class);
+        Class<?>[] testClasses = config.getValue("tck.config.test.javaconfig.converter.class.array", Class[].class);
+        assertEquals(testClasses.length, 2);
+        assertEquals(testClasses, new Class[]{ClassConverterTest.class, String.class});
+    }
+
+    @Test
+    public void testConverterForClassLoadedInBean() {
+        assertEquals(classConverterBean.testClass, ClassConverterTest.class);
+        assertEquals(classConverterBean.testClasses.length, 2);
+        assertEquals(classConverterBean.testClasses, new Class[]{ClassConverterTest.class, String.class});
+        assertEquals(classConverterBean.testClassSet.size(), 2);
+        assertEquals(classConverterBean.testClassSet, new LinkedHashSet<>(Arrays.asList(ClassConverterTest.class, String.class)));
+        assertEquals(classConverterBean.testClassList.size(), 2);
+        assertEquals(classConverterBean.testClassList, Arrays.asList(ClassConverterTest.class, String.class));
+    }
+
+    @Dependent
+    private static class ClassConverterBean {
+        @Inject
+        @ConfigProperty(name = "tck.config.test.javaconfig.converter.class")
+        private Class<?> testClass;
```

Comment:
@johnament Do you have an implementation that passes the test?

In [http://github.com/wildfly-extras/wildfly-microprofile-config](wildfly-microprofile-config) that is using Weld to run the TCK, the test fails with:

```
WELD-001408: Unsatisfied dependencies for type Class<?>[] with qualifiers @ConfigProperty
  at injection point [BackedAnnotatedField] @Inject @ConfigProperty private org.eclipse.microprofile.config.tck.ClassConverterTest$ClassConverterBean.testClasse
s
  at org.eclipse.microprofile.config.tck.ClassConverterTest$ClassConverterBean.testClasses(ClassConverterBean.java:0)
```

If I update the test to remote the wildcard from all injected types (using instead `private Class testClass`), the test passes.

Reply:
I cannot get this test pass either.